ycmd.el

```diff
@@ -610,26 +610,30 @@ control.) The newly started server will have a new HMAC secret."
 
   (ycmd--start-keepalive-timer))
 
+(defun ycmd--shutdown ()
+  "Send a shutdown request to the server."
+  (deferred:sync!
```

Comment:
Yeah, that seems right. Otherwise we'd have to make `ycmd-close` use deferreds, and I don't see a good reason to do that.